src/System.CommandLine.NamingConventionBinder/ModelBinder.cs

```diff
@@ -312,12 +312,26 @@ private bool UpdateInstanceInternalNotifyIfNonDefaultsUsed<T>(T instance, Bindin
                 return (new BoundValue(parameterDescriptor.GetDefaultValue(), valueDescriptor, valueSource), false);
             }
 
-            return (BoundValue.DefaultForValueDescriptor(valueDescriptor), false);
+            return (DefaultForValueDescriptor(valueDescriptor, bindingContext), false);
         }
 
         return (null, false);
     }
 
+    private static BoundValue DefaultForValueDescriptor(
+        IValueDescriptor valueDescriptor,
+        BindingContext context)
+    {
+        var valueSource = ValueDescriptorDefaultValueSource.Instance;
+
+        valueSource.TryGetValue(valueDescriptor, context, out var value);
```

Comment:
Should we care about the case where `TryGetValue` returns false? Should there be something line `BoundValue.None` to return in that case?